src/main/frontend/components/select.cljs

```diff
@@ -0,0 +1,95 @@
+  "Generic component for fuzzy searching items to select an item. See
+  select-config to add a new use or select-type for this component. To use the
+  new select-type, set :ui/open-select to the select-type. See
+  :select-graph/open command for an example."
+  (:require [frontend.modules.shortcut.core :as shortcut]
+            [frontend.context.i18n :as i18n]
+            ;; Would prefer to depend on handler.repo but unable to because
+            ;; of a circular dependency
+            [frontend.components.repo :as repo]
+            [frontend.search :as search]
+            [frontend.state :as state]
+            [frontend.ui :as ui]
+            [frontend.util :as util]
+            [rum.core :as rum]
+            [frontend.config :as config]
+            [reitit.frontend.easy :as rfe]))
+
+(rum/defc render-item
+  [{:keys [id value]} chosen?]
+  [:div.inline-grid.grid-cols-4.gap-x-4.w-full
+   {:class (when chosen? "chosen")}
+   [:span.col-span-3 value]
+   [:div.col-span-1.justify-end.tip.flex
+    (when id
+      [:code.opacity-20.bg-transparent id])]])
+
+(rum/defcs select <
+  (shortcut/disable-all-shortcuts)
+  (rum/local "" ::input)
+  {:will-unmount (fn [state]
+                   (state/set-state! [:ui/open-select] false)
+                   state)}
+  [state {:keys [items limit on-chosen empty-placeholder]
+          :or {limit 100}}]
+  (rum/with-context [[t] i18n/*tongue-context*]
+    (let [input (::input state)]
+      [:div.cp__select.cp__select-main
+       [:div.input-wrap
+        [:input.cp__select-input.w-full
+         {:type        "text"
+          :placeholder (t :select/prompt)
+          :auto-focus  true
+          :value       @input
+          :on-change   (fn [e] (reset! input (util/evalue e)))}]]
+
+       [:div.item-results-wrap
+        (ui/auto-complete
+         (search/fuzzy-search items @input :limit limit :extract-fn :value)
+         {:item-render render-item
+          :class       "cp__select-results"
+          :on-chosen   (fn [x]
+                         (state/close-modal!)
+                         (on-chosen x))
+          :empty-placeholder empty-placeholder})]])))
+
+(defn select-config
+  "Config that supports multiple types (uses) of this component. To add a new
+  type, add a key with the value being a map with the following keys:
+
+  * :items-fn - fn that returns items with a :value key that are used for the
+    fuzzy search and selection. Items can have an optional :id and are displayed
+    lightly for a given item.
+  * :on-chosen - fn that is given item when it is chosen."
+  []
+  {:select-graph
+   {:items-fn (fn []
+                (->>
+                 (state/get-repos)
+                 (remove (fn [{:keys [url]}]
+                           (or (config/demo-graph? url)
+                               (= url (state/get-current-repo)))))
+                 (map (fn [{:keys [url]}]
+                        (hash-map :value (second (util/get-dir-and-basename url))
```

Comment:
Breaks the browser app.

https://github.com/logseq/logseq/blob/ca755f26052c8bfa3abbeae1e239c36288f45959/src/main/frontend/components/repo.cljs#L94-L98

![image](https://user-images.githubusercontent.com/72891/150808233-cd8d5bf5-49d6-4854-958e-093966e3841e.png)

Reply:
Thanks! I haven't used the browser app much yet. Taking a look

Reply:
The switching works but the value/name on the left is confusing. I'll update it so that left side is "bugs" and right side is still the same